src/client/cli/cmd/launch.cpp

```diff
@@ -396,20 +397,22 @@ mp::ParseCode cmd::Launch::parse_args(mp::ArgParser* parser)
             {
                 auto file_type = fs::status(cloudInitFile.toStdString()).type();
                 if (file_type != fs::file_type::regular && file_type != fs::file_type::fifo)
-                {
-                    cerr << "error: No such file: " << cloudInitFile.toStdString() << "\n";
-                    return ParseCode::CommandLineError;
-                }
+                    throw YAML::BadFile{};
 
                 node = YAML::LoadFile(cloudInitFile.toStdString());
             }
             request.set_cloud_init_user_data(YAML::Dump(node));
         }
-        catch (const std::exception& e)
+        catch (const YAML::BadFile& e)
         {
-            cerr << "error loading cloud-init config: " << e.what() << "\n";
+            auto err_detail = fmt::format("{}\n{}", e.what(), "Please ensure that Multipass can read it.");
+            fmt::println(cerr, err_msg_template, cloudInitFile, err_detail);
             return ParseCode::CommandLineError;
         }
+        catch (const YAML::Exception& e)
+        {
+            fmt::println(cerr, err_msg_template, cloudInitFile, e.what());
```

Comment:
Shouldn't this also return `ParseCode::CommandLineError`? Now, if the yaml file is invalid, the error gets printed, but the launch continues.

Reply:
Yes, duh :facepalm: